src/erlcloud_kms.erl

```diff
@@ -0,0 +1,1060 @@
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% KMS API:
+%% [http://docs.aws.amazon.com/kms/latest/APIReference/API_UpdateKeyDescription.html]
+%%
+%% ===Example===
+%%
+%%------------------------------------------------------------------------------
+-spec update_key_description/2 ::
+          (KeyId :: string(),
+           Description :: string()) ->
+          any().
+update_key_description(KeyId, Description) ->
+    update_key_description(KeyId, Description, default_config()).
+
+
+-spec update_key_description/3 ::
+          (KeyId :: string(),
+           Description :: string(),
+           Config :: aws_config()) ->
+          any().
+update_key_description(KeyId, Description, Config) ->
+    Json = [{<<"KeyId">>, KeyId}, {<<"Description">>, Description}],
+    request(Config, "TrentService.UpdateKeyDescription", Json).
+
+
+-spec kms_result_fun/1 ::
+          (Request :: aws_request()) -> aws_request().
+kms_result_fun(#aws_request{response_type = ok} = Request) ->
+    Request;
+kms_result_fun(#aws_request{response_type = error, 
+                           error_type = aws, 
+                           response_status = Status} = Request)
+  when Status >= 500 ->
+    Request#aws_request{should_retry = true};
+kms_result_fun(#aws_request{response_type = error, error_type = aws} = Request) ->
+    Request#aws_request{should_retry = false}.
+
+
+%
+% Private
+%
+request(Config, Operation, Body) ->
+    Payload = case Body of
+               [] -> <<"{}">>;
+               _ -> jsx:encode(Body)
+           end,
+    Headers = headers(Config, Operation, Payload),
+    Request = #aws_request{service = kms,
+                           uri = uri(Config),
+                           method = post,
+                           request_headers = Headers,
+                           request_body = Payload},
+    case erlcloud_aws:request_to_return(erlcloud_retry:request(Config, Request, fun kms_result_fun/1)) of
+        {ok, {_RespHeaders, RespBody}} -> {ok, jsx:decode(RespBody)};
+        {error, _} = Error-> Error
+    end.
+
+
+headers(Config, Operation, Body) ->
+    Headers = [{"host", Config#aws_config.kms_host},
+               {"x-amz-target", Operation},
+               {"content-type", "application/x-amz-json-1.1"}],
+    Region = erlcloud_aws:aws_region_from_host(Config#aws_config.kms_host),
+    erlcloud_aws:sign_v4_headers(Config, Headers, Body, Region, "kms").
+
+
+uri(#aws_config{kms_scheme = Scheme, kms_host = Host} = Config) ->
+    lists:flatten([Scheme, Host, port_spec(Config)]).
+
+
+port_spec(#aws_config{kms_port=80}) ->
+    "";
+port_spec(#aws_config{kms_port=Port}) ->
+    [":", erlang:integer_to_list(Port)].
+
+
+dynamize_options(List) ->
+    dynamize_options(List, []).
+
+
+dynamize_options([{Key, Value}|T], Acc) ->
+    case dynamize_option_key(Key) of
+        undefined -> dynamize_options(T, Acc);
```

Comment:
I would like the library to validate the inputs and not just throw away unknowns silently. Maybe throw an exception or return  `{error, wrong_opt}` here?